TROUBLESHOOTING.md

```diff
@@ -72,7 +72,7 @@ The data generation step is executed via the `ilab data generate` command, and i
 1. Increase the number of instructions generated by passing the `--num-instructions` flag to the `ilab data generate` command as follows: `ilab data generate --num-instructions 1000`.
    The `--num-instructions` flag will generate 1000 points of synthetic data based on your provided examples. The greater the number of instructions generated, the better the model will be trained (within reasonable limits).
 
-3. Using a better model via `--model`. Larger models can lead to better data generation. This option requires users to be familiar with various existing models, and which specific models would suit their needs. This could mean either using a model with more nodes than the default InstructLab `merlinite-7b-lab` model, such as the `Mixtral-8x7B-Instruct-v0.1` model, or using an unquantized version of the InstructLab `merlinite-7b-lab` model. It can be used as follows: `ilab serve --model-path models/mixtral-8x7b-instruct-v0.1.Q4_K_M.gguf` and `ilab data generate --model models/mixtral-8x7b-instruct-v0.1.Q4_K_M.gguf`
+3. Using a better model via `--model`. Larger models can lead to better data generation. This option requires users to be familiar with various existing models, and which specific models would suit their needs. This could mean either using a model with more nodes than the default InstructLab `granite-7b-lab` model, such as the `Mixtral-8x7B-Instruct-v0.1` model, or using an unquantized version of the InstructLab `granite-7b-lab` model. It can be used as follows: `ilab serve --model-path models/mixtral-8x7b-instruct-v0.1.Q4_K_M.gguf` and `ilab data generate --model models/mixtral-8x7b-instruct-v0.1.Q4_K_M.gguf`
```

Comment:
one question I had: is it ok to switch the teacher model to granite? The idea behind the `simple` teacher being merlinite is that it is "smarter" than granite, part of me worries this will degrade the simple pipeline results further

Reply:
I think we can leave the default teacher model to be `merlinite` - my only concern with it is the default model downloads - would we then be downloading three default models?
1. Merlinite GGUF as the teacher
2. Granite GGUF as the student
3. Mistral GGUF for SDG
If we're fine with this, would appreciate some pointers as to where exactly I need to make the changes to implement what I've laid out here

Reply:
We would definitely need to download at least 2 models - the teacher and the student, which need to be different for the sake of the student being able to catch up to the teacher. What you could do is make the mistral GGUF optional, and maybe down the road provide a way for the user to select which one they'd like.

Reply:
I think currently we are using Merlinite by default as both the student and the teacher and Mistral for SDG? I could be wrong, Charlie and @jaideepr97 would know for sure

Irregardless I'm happy to change this up a bit, but I don't want to stray too far from the goal of this PR

Reply:
> student and the teacher and Mistral for SDG?

The teacher would be the model being used for SDG, so if it's teaching itself then that is a big problem.

Reply:
Oh I may be mistaken then, the current teacher might be Mistral? But in that case I'm not sure what @cdoern is referring to - am I accidentally switching it from Mistral to Granite? Because that was definitely not my intention

Reply:
@nathan-weinberg You are correct, Merlinite is the version of Mistral-7B trained by IBM research. Granite is a separate foundation model from IBM. 

What we need to do is have the teacher model be more capable, which in the case of the simple pipeline has generally been the case. If we're trimming down on models that are downloaded, then we need to be sure that we still hold this property. I'm unfamiliar with the simple pipeline, but I don't believe we would be using base Mistral as a teacher model for SDG, so that may be the one to remove (we need to confirm this). 

But this pipeline is overall fairly old now, and IBM has Granite-3.0 8B and 3B, which could be potential replacements for either teacher & student models. We would just need to run some experiments to better understand the performance difference.

Reply:
ok so there seems to be a bit of confusion here. Here is the current state of SDG upstream:

pipeline: full, model: mistral 7b instruct GGUf

pipeline: simple, model: merlinite 7b lab GGUF

My concern here is that in the docs and the defaults it seems that merl is being replaced w/ granite in SDG. I think we should:

a. download granite gguf as the default chat model
b. download merl gguf as the simple teacher model
c. download mistral gguf as the full teacher model

for _now_ we can download the all. However, I think we should conditionally download merlinite or mistral depending on the pipeline in the config eventually.

Reply:
@cdoern Why is mistral GGUF being used as the full teacher model for the full pipeline while Merlinite is the full teacher model for the simple pipeline? Merlinite is produced as a result of fine-tuning Mistral, so there should ideally only be one model being used for this.

Reply:
Regardless of which models we're using, it seems like an easy solution here would be to have the following flow:

1. At `ilab init` the user would be prompted if they want to download all of the models now, or defer it until they need it

2. Then when a defined model doesn't exist on disk, or if they go to run SDG and the teacher models aren't present, they are prompted with a message like: "Mistral-7B-GGUF is required for this pipeline but appears to be missing. Would you like to download it?" And if they hit no, the pipeline is cancelled, else it's downloaded.

Reply:
@RobotSail Merlinite does not work with the full pipeline, consistently yields 0 generated samples and/or times out on the generation

Reply:
This PR is simply switching the default model to granite. 

I don't think we should make code changes outside of maybe downloading a third model, can we re-visit the download syntax in another issue/PR

Reply:
I'm going to go with this for now, @RobotSail i do wanna keep this convo going somewhere, either in Slack or a new issue, do you have a preference?

Reply:
Yes, we'll make an issue/epic to describe the overall procedure.